eec-backend/Services/CalculationService.cs

```diff
@@ -20,15 +20,52 @@ public CalculationService(IProductService productService)
 
         public async Task<BaseResponse> GetCalculation(BaseRequest baseRequest)
         {
+            
             Product product = await _productService.GetProductById(baseRequest.ModelIdentifier);
-
-            BaseResponse baseResponse = new()
+            BaseResponse response = new BaseResponse();
+            response.Given = new SingleCalculationModel()
             {
                 AnnualCost = GetAnnualCost(product, baseRequest),
                 EcoScore = GetEcoScore(product),
+                Class = product.EnergyEfficiencyClass,
+                Category = product.Category,
+                Index = product.EnergyEfficiencyIndex,
+                Manufacturer = product.SupplierOrTrademark,
+                ModelIdentifier = product.ModelIdentifier
             };
             
-            return baseResponse;
+            var recc = ReccommendedProducts(product);
+            Dictionary<string, List<SingleCalculationModel>> upgrades = new();
+
+            foreach(var item in recc)
+            {
+                upgrades.Add(item.Key, new List<SingleCalculationModel>
+                    (item.Value.Select(x => new SingleCalculationModel()
+                    {
+                        AnnualCost = GetAnnualCost(x, baseRequest),
+                        EcoScore = GetEcoScore(x),
+                        Class = x.EnergyEfficiencyClass,
+                        Category = x.Category,
+                        Index = x.EnergyEfficiencyIndex,
+                        Manufacturer = x.SupplierOrTrademark,
+                        ModelIdentifier = x.ModelIdentifier
+                    })).ToList());
+            }
+
+            response.Upgrades = upgrades;
+
+            return response;
+        }
+        
+        private Dictionary<string, List<Product>> ReccommendedProducts(Product product)
```

Comment:
Typo in 'Recommended' (should be single 'c')

Reply:
Thanks, done